doc/architecture-decisions/0012-stopped-app-routing.md

```diff
@@ -0,0 +1,66 @@
+# 12. Stopped App Routing
+
+Date: 2020-05-29
+
+## Status
+
+Proposal
+
+## Context
+
+In CF-for-VMs, when a route is mapped to multiple apps, some started and some
+stopped, gorouter only directs requests to the apps that are started.
+
+In CF-for-k8s _currently_, we balance traffic equally between the started and
+stopped apps. This results in 503 errors for the user, and is not feature
+pairity.
+
+This is because the components that create the Istio configuration do not
+currently know about the status of apps. They assume all destinations on a Route
+object are up and able to receive traffic.
+
+See this issue: https://github.com/cloudfoundry/cf-for-k8s/issues/158
+
+## Proposal
+
+We have two proposals:
+
+1. Cloud Controller only adds destinations to Routes that are running. This way
+   Route objects remain the single source of truth for routing in the platform.
+   * _Note:_ Routes with _no_ destinations are explicitly supported already.
+
+2. Cloud Controller writes an App CR (this is a new CRD) that contains the
```

Comment:
This second proposal seems like a whole lot of implementation on CAPI's end to figure out if the application instance is started or not. I feel like while both implementations are mutually exclusive, the first proposal can be an evolutionary step to the second proposal, at least until there are more use cases for holding all of the application information on an App CR.

Reply:
Totally agree. I think an App CR is an eventual goal for CAPI anyway, so this could be some evidence for prioritization of that. If they were planning to do it sooner rather than later we could tack this onto that effort.

Personally I strongly prefer proposal 1 because it's the least work for everyone, and the contents of Route are always true.